server/datastore/mysql/app_configs.go

```diff
@@ -23,6 +29,32 @@ func (d *Datastore) AppConfig() (*kolide.AppConfig, error) {
 	return info, nil
 }
 
+func (d *Datastore) ManageHostExpiryEvent(hostExpiryEnabled bool, hostExpiryWindow int) error {
+	rows, err := d.db.Query("SELECT @@event_scheduler")
+	if err != nil {
+		return err
+	}
+	if !rows.Next() {
+		return errors.New("Error detecting event scheduler status.")
+	}
+	var value string
+	if err := rows.Scan(&value); err != nil {
+		return err
+	}
+	if value != "ON" {
+		return errors.New("Event Scheduler must be enabled for configuring Host Expiry.")
+	}
+
+	if !hostExpiryEnabled {
```

Comment:
Definitely an edge case, but it could be good to move this above the check for the event scheduler so that a user can disable this setting after the event scheduler has been turned off on their MySQL instance.